### ISSUE TITLE
Serve http and socks5 on the same port

### ISSUE THREAD

**Issue by raintears2** [z3APA3A/3proxy#616]
Will 3proxy support http and socks5 on the same port in the develop version?
I found this proxy software (https://github.com/nadoo/glider) can do this.I tested http and socks5,it works  on the same port.

**Comment by z3APA3A** [comment]
No, 3proxy requires explicit service to be allocated to port. It's possible and relatively easy to make an auto-detection, but can you explain your usecase somehow, because, at first look, it seems pretty useless?

**Comment by raintears2** [comment]
One port make proxy more simple and easy to use for clients.One port can be used in all kinds of client software,we don't need to know which proxy type they can support(some of them just support one,http or socks5 only).

Proxy often provide both socks5 and http.With two ports,we have to store them in two proxy files.
Some beginner maybe choose the wrong proxy type and can't connect successfully.
One port can avoid these situation and reduce the difficulty of using proxy.

As you say,it is relatively easy to make an auto-detection.Could you show me where to do this detection?

Kind regards
Rain

**Comment by atimofeev86** [comment]
Hello. Is this still not working?

**Comment by raintears2** [comment]
Nice work!!!
It's amazing.I will test it soon.

**Comment by atimofeev86** [comment]
it's worked. thank you!
but in logs I see empty %N (%N - Service name (PROXY, SOCKS, POP3P, etc))

**Comment by zbutfly** [comment]
Would anybody please give a compiled ball including this feature? I run it under Windows x64.
